src/org/labkey/test/tests/DomainFieldTypeChangeTest.java

```diff
@@ -0,0 +1,205 @@
+    }
+
+    @Test
+    public void testNonProvisionedDomainFieldChanges()
+    {
+        String assayName = "Assay1";
+        String runName = "Run1";
+        File runFile = new File(TestFileUtils.getSampleData("AssayImportExport"), "GenericAssay_Run1.xls");
+        goToManageAssays();
+        APIAssayHelper assayHelper = new APIAssayHelper(this);
+        ReactAssayDesignerPage assayDesignerPage = assayHelper.createAssayDesign("General", assayName);
+
+        DomainFormPanel runFields = assayDesignerPage.goToRunFields();
+        runFields.addField("runTestInteger").setType(FieldDefinition.ColumnType.Integer);
+        runFields.addField("runTestDecimal").setType(FieldDefinition.ColumnType.Decimal);
+        runFields.addField("runTestDate").setType(FieldDefinition.ColumnType.DateAndTime);
+        runFields.addField("runTestBoolean").setType(FieldDefinition.ColumnType.Boolean);
+
+        DomainFormPanel batchFields = assayDesignerPage.goToBatchFields();
+        batchFields.addField("batchTestInteger").setType(FieldDefinition.ColumnType.Integer);
+        batchFields.addField("batchTestDecimal").setType(FieldDefinition.ColumnType.Decimal);
+        batchFields.addField("batchTestDate").setType(FieldDefinition.ColumnType.DateAndTime);
+        batchFields.addField("batchTestBoolean").setType(FieldDefinition.ColumnType.Boolean);
+        assayDesignerPage.clickFinish();
+
+        goToManageAssays();
+        clickAndWait(Locator.linkWithText(assayName));
+        DataRegionTable table = new DataRegionTable("Runs", getDriver());
+        table.clickHeaderButton("Import Data");
+        setFormElement(Locator.name("batchTestInteger"), "1");
+        setFormElement(Locator.name("batchTestDecimal"), "1.1");
+        setFormElement(Locator.name("batchTestDate"), "01-01-2022");
+        checkCheckbox(Locator.name("batchTestBoolean"));
+        clickButton("Next");
+
+        setFormElement(Locator.name("name"), runName);
+        setFormElement(Locator.name("runTestInteger"), "12");
+        setFormElement(Locator.name("runTestDecimal"), "1.12");
+        setFormElement(Locator.name("runTestDate"), "01-03-2022");
+        checkRadioButton(Locator.radioButtonById("Fileupload"));
+        setFormElement(Locator.input("__primaryFile__"), runFile);
+        clickButton("Save and Finish");
+
+        goToManageAssays();
+        clickAndWait(Locator.linkWithText(assayName));
+        waitForElement(Locator.linkWithText(runName));
+        click(Locator.linkWithText("Manage assay design"));
+        assayDesignerPage = _assayHelper.clickEditAssayDesign();
+        runFields = assayDesignerPage.goToRunFields();
+        runFields.getField("runTestInteger").setType(FieldDefinition.ColumnType.Decimal);
+        checker().verifyEquals("Incorrect error message for integer to decimal", Arrays.asList("Cannot convert from INTEGER to DOUBLE for non-provisioned table"), assayDesignerPage.clickSaveExpectingErrors());
```

Comment:
This should be supported. I can see the problem in the code. Can you log a bug please?